Update Example (#9)
DevExpress-Examples/blazor-theme-and-size-mode-switcher: Readme.md

<!-- default badges end -->
# How to implement a Theme Switcher in Blazor applications

> This example demonstrates how to create a Theme Switcher as in [DevExpress Blazor Demos](https://demos.devexpress.com/blazor/) and apply the selected theme to an application at runtime. The Theme Switcher includes the standard Bootstrap theme, [DevExpress built-in themes](https://github.com/DevExpress/bootstrap-themes), and [free Bootswatch themes](https://bootswatch.com/).

Reviewer: ```suggestion
This example demonstrates how you can add a Theme Switcher to your application. The switcher in this example is the same as in [DevExpress Blazor Demos](https://demos.devexpress.com/blazor/). The choices will include [DevExpress built-in themes](https://github.com/DevExpress/bootstrap-themes) and Bootstrap themes (the default theme and [free Bootswatch options](https://bootswatch.com/)).

```

<!-- default badges end -->
# How to implement a Theme Switcher in Blazor applications

This example demonstrates how you can add a Theme Switcher to your application. The switcher in this example is the same as in [DevExpress Blazor Demos](https://demos.devexpress.com/blazor/). The choices will include [DevExpress built-in themes](https://github.com/DevExpress/bootstrap-themes) and Bootstrap themes (the default theme and [free Bootswatch options](https://bootswatch.com/)).

![Blazor - Theme Switcher](images/blazor-theme-switcher.png)

> The example includes solutions both for Blazor Server and Blazor WebAssembly hosting models. Note that these solutions target .NET 6.0.

Reviewer 2: ```suggestion
The example includes solutions for both Blazor Server and Blazor WebAssembly hosting models. Note that these solutions target .NET 6.0.
```

    * Event callbacks (`VisibleChanged` and `ActiveThemeChanged`) that update the state of the parent component after you select another theme. Refer to the Microsoft documentation for more information: [Binding with Component Parameters](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/data-binding#binding-with-component-parameters).
    * A theme collection that stores all themes available in the Theme Switcher (`GroupedThemes`).
   
    ```razor
    @* ... *@
    @code {
        static readonly Dictionary<string, string[]> GroupedThemes = new() {
            ["DevExpress Themes"] = Utils.DevExpressThemes,
            ["Bootstrap Themes"] = new[] {
                "default", "cerulean", "cyborg", "flatly", "journal", "litera", "lumen", "lux", "pulse", "simplex", "solar", "superhero", "united", "yeti"
            }
        };
     
        [Parameter] public bool Visible { get; set; }
        [Parameter] public EventCallback<bool> VisibleChanged { get; set; }
        [Parameter] public ThemeItem ActiveTheme { get; set; }
        [Parameter] public EventCallback<ThemeItem> ActiveThemeChanged { get; set; }
        @* ... *@
    }
    ```

> 7. To display icons for each theme in the Theme Switcher, add the sidebar's layout into the *ThemeSwitcherSideView.razor* file:

Reviewer 2: ```suggestion
7. To display icons for each theme in the Theme Switcher, add the sidebar's layout to the *ThemeSwitcherSideView.razor* file:

```

    @page "/"
    @using switcher.Data
    @inject WeatherForecastService ForecastService
        
    <DxGrid Data="@forecasts"
            PageSize="5"
            ShowFilterRow="false"
            @bind-SelectedDataItems="@SelectedDataItems">
        <Columns>
            <DxGridSelectionColumn />
            <DxGridDataColumn FieldName="@nameof(WeatherForecast.TemperatureC)" Caption="@("Temp. (\x2103)")" TextAlignment="@GridTextAlignment.Right"></DxGridDataColumn>
            <DxGridDataColumn FieldName="@nameof(WeatherForecast.TemperatureF)" Caption="@("Temp. (\x2109)")"></DxGridDataColumn>
            <DxGridDataColumn FieldName="@nameof(WeatherForecast.Date)" DisplayFormat="D"></DxGridDataColumn>
        </Columns>
    </DxGrid>
    
    @code {
        WeatherForecast[] forecasts;
        IReadOnlyList<object> SelectedDataItems { get; set; }
    
        protected override async Task OnInitializedAsync() {
            forecasts = await ForecastService.GetForecastAsync(DateTime.Today);
            SelectedDataItems = forecasts.Where(f => f.Date == DateTime.Today.AddDays(1)).ToList();       
            System.Threading.Thread.Sleep(3000);
        }
    }
    
    ```

> This example uses the `HeadOutlet` component. Refer to the following Microsoft article for more information: [HeadOutlet Component](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content#headoutlet-component).

Reviewer 2: ```suggestion
This example uses the `HeadOutlet` component. Refer to the following section of the **Control <head> content in ASP.NET Core Blazor apps** Microsoft article for more information: [HeadOutlet component](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content#headoutlet-component).
```
